homeassistant/components/reolink/binary_sensor.py

```diff
@@ -71,7 +71,16 @@ class ReolinkBinarySensorEntityDescription(BinarySensorEntityDescription):
         icon="mdi:dog-side",
         icon_off="mdi:dog-side-off",
         value=lambda api, ch: api.ai_detected(ch, PET_DETECTION_TYPE),
-        supported=lambda api, ch: api.ai_supported(ch, PET_DETECTION_TYPE),
+        supported=lambda api, ch: api.ai_supported(ch, PET_DETECTION_TYPE)
+        and not api.supported(ch, "ai_animal"),
+    ),
+    ReolinkBinarySensorEntityDescription(
+        key=PET_DETECTION_TYPE,
```

Comment:
Is it intended that this is the same key as in the description above?

Reply:
Yes, because only one of the two can be present.
It is basically the same feature: Animal detection is the extended version of Pet detection.
This is done so users on older firmwares that have Pet detection will not need to change there entities when upgrading there firmware to get animal detection instead of pet detection.

Reply:
Okay, perfect, I just wanted to be sure, but I suspected something like this 👍